### PR TITLE
Make server installation FHS compilant

### LINKED ISSUE #26590: Make Framework and API's installation FHS compliant.
## Description

We need to continue the work done in #26349 making the installation for the server and the API's compatible with the [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.html) standard. 

The problem lies in the current way of defining the `WAZUH_PATH`, which is based on searching the framework `directory`. This will soon be obsolete, and we need to define different paths for each purpose.

The proposed hierarchy is:

* Binary files (`wazuh-server`, `wazuh-apid`, `wazuh-comms-apid`) :arrow_right:  `/bin`
* Embedded Python, API's, and server scripts, `installation_uuid` :arrow_right:  `/usr/share/wazuh-server`
* Configuration file (`wazuh-server.yml`), SSL certificates, PEM keys :arrow_right:  `/etc/wazuh-server`
* Cluster and Comms API sockets, PID files :arrow_right:  `/run/wazuh-server`
* Log files :arrow_right:  `/var/log/wazuh-server`
* RBAC DB :arrow_right:  `/var/lib/wazuh-server`

## Plan

- [x] Define the new paths at `framework/wazuh/core/common.py`
- [x] Replace the occurrence of `WAZUH_PATH` with the new paths depending on the case
- [x] Fix the related unit tests after the replacement
- [x] Modify the `src/init/inst-functions.sh` to create the new directories if it does not exist previously.
- [x] Modify the `package/debs/SPECS/debian/rules` and `packages/rpms/SPECS/wazuh-server.spec` to include the new package hierarchy.

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|---|
|#26590|

## Description

This PR closes #26590. Adapts the wazuh server installation to be FHS compliant.

Builds:
* https://github.com/wazuh/wazuh/actions/runs/11786792370
* https://github.com/wazuh/wazuh/actions/runs/11786796690

## Logs/Alerts example

<details><summary>Master Startup</summary>
<p>

```console
root@wazuh-master:/# wazuh-server start -rf
Starting cluster in foreground (pid: 69304)
2024/11/11 22:01:36 INFO: [Cluster] [Main] Started wazuh-engined (pid: 69305)
2024/11/11 22:01:36 INFO: [Cluster] [Main] Started wazuh-apid (pid: 69306)
2024/11/11 22:01:36 INFO: [Cluster] [Main] Started wazuh-comms-apid (pid: 69307)
2024-11-11 22:01:36.747 69305:69305 info: Logging initialized.
2024-11-11 22:01:36.749 69305:69305 info: Store initialized.
2024-11-11 22:01:36.751 69305:69305 info: RBAC initialized.
2024-11-11 22:01:36.752 69305:69305 info: MetricsManager: Created new scope: (KVDB)
2024-11-11 22:01:36.769 69305:69305 info: KVDB initialized.
2024-11-11 22:01:36.773 69305:69305 info: Geo initialized.
2024-11-11 22:01:36.788 69305:69305 info: Schema initialized.
2024/11/11 22:01:37 INFO: [Local Server] [Main] Serving on /run/wazuh-server/socket/c-internal.sock
2024/11/11 22:01:37 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
2024/11/11 22:01:37 INFO: [Master] [Local integrity] Starting.
2024/11/11 22:01:37 INFO: [Master] [Local integrity] Finished in 0.099s. Calculated metadata of 2 files.
2024-11-11 22:01:37.526 69305:69305 info: Loaded timezone database version: '2024a'
2024-11-11 22:01:37.531 69305:69305 info: HLP initialized.
2024-11-11 22:01:37.543 69305:69305 info: Builder initialized.
2024-11-11 22:01:37.543 69305:69305 info: Catalog initialized.
2024-11-11 22:01:37.543 69305:69305 info: Policy manager initialized.
2024-11-11 22:01:37.543 69305:69305 info: MetricsManager: Created new scope: (EventQueue)
2024-11-11 22:01:37.543 69305:69305 info: MetricsManager: Created new scope: (EventQueueDelta)
2024-11-11 22:01:37.544 69305:69305 info: The queue will be flooded in the file: /var/ossec/logs/engine-flood.log
2024-11-11 22:01:37.545 69305:69305 info: MetricsManager: Created new scope: (TestQueue)
2024-11-11 22:01:37.545 69305:69305 info: MetricsManager: Created new scope: (TestQueueDelta)
2024-11-11 22:01:37.553 69305:69305 info: No flooding file provided, the queue will not be flooded.
2024-11-11 22:01:37.554 69305:69305 warning: Router: router/router/0 table is empty
2024-11-11 22:01:37.558 69305:69305 info: Router: router/tester/0 table not found in store. Creating new table: File '/var/ossec/engine/store/router/tester/0' could not be parsed: JSON document could not be parsed: The document is empty.
2024-11-11 22:01:37.558 69305:69305 info: Router initialized.
2024-11-11 22:01:37.560 69305:69305 info: Starting database file decompression.
2024-11-11 22:01:37.569 69305:69305 error: Error opening the database: Couldn't find column family: 'vendor_map', trying to re-download the feed.
2024-11-11 22:01:37.570 69305:69305 info: MetricsManager: Created new scope: (endpointAPI)
2024-11-11 22:01:37.570 69305:69305 info: MetricsManager: Created new scope: (endpointAPIRate)
2024-11-11 22:01:37.571 69305:69305 info: MetricsManager: Created new scope: (endpointEvent)
2024-11-11 22:01:37.571 69305:69305 info: MetricsManager: Created new scope: (endpointEventRate)
2024-11-11 22:01:37.574 69305:69305 info: Starting the server...
2024/11/11 22:01:37 INFO: Starting API in foreground
2024/11/11 22:01:37 INFO: Starting API as root
2024/11/11 22:01:37 INFO: Checking RBAC database integrity...
2024/11/11 22:01:37 INFO: /var/lib/wazuh-server/rbac.db file was detected
2024/11/11 22:01:37 INFO: RBAC database integrity check finished successfully
2024/11/11 22:01:37 INFO: Starting API in foreground
2024/11/11 22:01:37 INFO: Starting API as root
2024/11/11 22:01:37 INFO: Listening on 0.0.0.0:27000
2024/11/11 22:01:38 INFO: Starting gunicorn 22.0.0
2024/11/11 22:01:38 INFO: Listening at: https://0.0.0.0:27000 (69307)
2024/11/11 22:01:38 INFO: Using worker: uvicorn.workers.UvicornWorker
2024/11/11 22:01:38 INFO: Booting worker with pid: 69461
2024/11/11 22:01:38 INFO: Booting worker with pid: 69462
2024/11/11 22:01:38 INFO: Started server process [69307]
2024/11/11 22:01:38 INFO: Waiting for application startup.
2024/11/11 22:01:38 INFO: Application startup complete.
2024/11/11 22:01:38 INFO: Uvicorn running on unix socket /run/wazuh-server/socket/comms-api.sock (Press CTRL+C to quit)
2024/11/11 22:01:38 INFO: Booting worker with pid: 69463
2024/11/11 22:01:38 INFO: Started server process [69463]
2024/11/11 22:01:38 INFO: Waiting for application startup.
2024/11/11 22:01:38 INFO: Application startup complete.
2024/11/11 22:01:38 INFO: Booting worker with pid: 69464
2024/11/11 22:01:38 INFO: Started server process [69464]
2024/11/11 22:01:38 INFO: Waiting for application startup.
2024/11/11 22:01:38 INFO: Application startup complete.
2024/11/11 22:01:40 INFO: Listening on 0.0.0.0:55000.
2024/11/11 22:01:40 INFO: Getting installation UID...
2024/11/11 22:01:40 INFO: Getting updates information...
2024/11/11 22:01:43 INFO: [Worker] [Main] Connection from ('172.18.0.5', 41828)
2024/11/11 22:01:45 INFO: [Master] [Local integrity] Starting.
2024/11/11 22:01:45 INFO: [Master] [Local integrity] Finished in 0.002s. Calculated metadata of 2 files.
2024/11/11 22:01:52 INFO: [Worker worker1] [Integrity check] Starting.
2024/11/11 22:01:52 INFO: [Worker worker1] [Integrity check] Finished in 0.004s. Received metadata of 2 files. Sync not required.
```

</p>
</details> 

<details><summary>Worker Startup</summary>
<p>

```console
root@wazuh-worker1:/# wazuh-server start -rf
Starting cluster in foreground (pid: 68471)
2024/11/11 22:01:43 INFO: [Cluster] [Main] Started wazuh-engined (pid: 68472)
2024/11/11 22:01:43 INFO: [Cluster] [Main] Started wazuh-apid (pid: 68473)
2024/11/11 22:01:43 INFO: [Cluster] [Main] Started wazuh-comms-apid (pid: 68474)
2024-11-11 22:01:43.357 68472:68472 info: Logging initialized.
2024-11-11 22:01:43.357 68472:68472 info: Store initialized.
2024-11-11 22:01:43.357 68472:68472 info: RBAC initialized.
2024-11-11 22:01:43.358 68472:68472 info: MetricsManager: Created new scope: (KVDB)
2024-11-11 22:01:43.369 68472:68472 info: KVDB initialized.
2024-11-11 22:01:43.369 68472:68472 info: Geo initialized.
2024-11-11 22:01:43.383 68472:68472 info: Schema initialized.
2024/11/11 22:01:43 INFO: [Local Server] [Main] Serving on /run/wazuh-server/socket/c-internal.sock
2024/11/11 22:01:43 INFO: [Worker worker1] [Main] Successfully connected to master.
2024-11-11 22:01:44.065 68472:68472 info: Loaded timezone database version: '2024a'
2024-11-11 22:01:44.068 68472:68472 info: HLP initialized.
2024-11-11 22:01:44.087 68472:68472 info: Builder initialized.
2024-11-11 22:01:44.087 68472:68472 info: Catalog initialized.
2024-11-11 22:01:44.087 68472:68472 info: Policy manager initialized.
2024-11-11 22:01:44.087 68472:68472 info: MetricsManager: Created new scope: (EventQueue)
2024-11-11 22:01:44.087 68472:68472 info: MetricsManager: Created new scope: (EventQueueDelta)
2024-11-11 22:01:44.089 68472:68472 info: The queue will be flooded in the file: /var/ossec/logs/engine-flood.log
2024-11-11 22:01:44.090 68472:68472 info: MetricsManager: Created new scope: (TestQueue)
2024-11-11 22:01:44.090 68472:68472 info: MetricsManager: Created new scope: (TestQueueDelta)
2024-11-11 22:01:44.100 68472:68472 info: No flooding file provided, the queue will not be flooded.
2024-11-11 22:01:44.101 68472:68472 warning: Router: router/router/0 table is empty
2024-11-11 22:01:44.101 68472:68472 warning: Router: router/tester/0 table is empty
2024-11-11 22:01:44.101 68472:68472 info: Router initialized.
2024-11-11 22:01:44.101 68472:68472 info: Starting database file decompression.
2024-11-11 22:01:44.111 68472:68472 error: Error opening the database: Couldn't find column family: 'vendor_map', trying to re-download the feed.
2024-11-11 22:01:44.111 68472:68472 info: MetricsManager: Created new scope: (endpointAPI)
2024-11-11 22:01:44.111 68472:68472 info: MetricsManager: Created new scope: (endpointAPIRate)
2024-11-11 22:01:44.113 68472:68472 info: MetricsManager: Created new scope: (endpointEvent)
2024-11-11 22:01:44.113 68472:68472 info: MetricsManager: Created new scope: (endpointEventRate)
2024-11-11 22:01:44.116 68472:68472 info: Starting the server...
2024/11/11 22:01:44 INFO: Starting API in foreground
2024/11/11 22:01:44 INFO: Starting API as root
2024/11/11 22:01:44 INFO: Checking RBAC database integrity...
2024/11/11 22:01:44 INFO: /var/lib/wazuh-server/rbac.db file was detected
2024/11/11 22:01:44 INFO: RBAC database integrity check finished successfully
2024/11/11 22:01:44 INFO: Starting API in foreground
2024/11/11 22:01:44 INFO: Starting API as root
2024/11/11 22:01:44 INFO: Listening on 0.0.0.0:27000
2024/11/11 22:01:44 INFO: Starting gunicorn 22.0.0
2024/11/11 22:01:44 INFO: Listening at: https://0.0.0.0:27000 (68474)
2024/11/11 22:01:44 INFO: Using worker: uvicorn.workers.UvicornWorker
2024/11/11 22:01:44 INFO: Booting worker with pid: 68639
2024/11/11 22:01:44 INFO: Started server process [68474]
2024/11/11 22:01:44 INFO: Waiting for application startup.
2024/11/11 22:01:44 INFO: Application startup complete.
2024/11/11 22:01:44 INFO: Uvicorn running on unix socket /run/wazuh-server/socket/comms-api.sock (Press CTRL+C to quit)
2024/11/11 22:01:44 INFO: Booting worker with pid: 68640
2024/11/11 22:01:44 INFO: Started server process [68640]
2024/11/11 22:01:44 INFO: Waiting for application startup.
2024/11/11 22:01:44 INFO: Application startup complete.
2024/11/11 22:01:44 INFO: Booting worker with pid: 68641
2024/11/11 22:01:44 INFO: Started server process [68641]
2024/11/11 22:01:44 INFO: Waiting for application startup.
2024/11/11 22:01:44 INFO: Application startup complete.
2024/11/11 22:01:44 INFO: Booting worker with pid: 68642
2024/11/11 22:01:44 INFO: Started server process [68642]
2024/11/11 22:01:44 INFO: Waiting for application startup.
2024/11/11 22:01:44 INFO: Application startup complete.
2024/11/11 22:01:47 INFO: Listening on 0.0.0.0:55000.
2024/11/11 22:01:52 INFO: [Worker worker1] [Integrity check] Starting.
2024/11/11 22:01:52 INFO: [Worker worker1] [Integrity check] Finished in 0.021s. Sync not required.
2024/11/11 22:02:01 INFO: [Worker worker1] [Integrity check] Starting.
2024/11/11 22:02:01 INFO: [Worker worker1] [Integrity check] Finished in 0.008s. Sync not required.

```

</p>
</details>

### CODE REVIEW

File: framework/wazuh/core/cluster/cluster.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "files":{
 
-        "etc/": {
+        "": {
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply framework/wazuh/core/cluster/cluster.json:4]: These paths end up concatenated with the root one. I can open an issue to review this behavior, I think we will exceed the scope of the issue if we do it now.

File: framework/wazuh/core/cluster/cluster.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
             "description": "JWT signing key pair"
         },
 
-        "etc/shared/": {
+        "shared": {
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply framework/wazuh/core/cluster/cluster.json:15]: Same here

File: framework/wazuh/core/common.py [hunk 3]
```diff
@@ -190,7 +191,29 @@ def get_context_cache() -> dict:
 _WAZUH_GID = None
 GROUP_NAME = 'wazuh'
 USER_NAME = 'wazuh'
-WAZUH_PATH = find_wazuh_path()
+WAZUH_PATH = ''
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply framework/wazuh/core/common.py:196]: I added a comment explaining the reason.

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by fdalmaup — APPROVED** [review]
LGTM

**Pull request merged**